packages/plugin-terraform/src/index.ts

```diff
@@ -100,29 +102,38 @@ export const plugin: PluginFunction<PluginConfig> = (
         const schema = tfg.resource('amplience_content_type_schema', name, {
           body: fn(
             'file',
-            `\${path.module}/schemas/${paramCase(node.name.value)}.json`
+            `\${path.module}/schemas/${paramCase(node.name.value)}${
+              schemaSuffix ? '-' + schemaSuffix : ''
+            }.json`
           ),
           schema_id: typeUri(node, hostname),
-          validation_level: isSlot ? 'SLOT' : 'CONTENT_TYPE'
+          validation_level: isSlot ? 'SLOT' : 'CONTENT_TYPE',
         })
 
         const shouldVisualize = maybeDirectiveValue<BooleanValueNode>(
           directive,
           'visualizations'
         )?.value
 
+        const iconUrl = maybeDirectiveValue<EnumValueNode>(
+          maybeDirective(node, 'icon')!,
```

Comment:
Where do you guarantee that icon exists?
I think you should surround this with a hasDirective

```ts
const iconUrl = hasDirective(node, 'icon') ? maybeDire....?.value : undefined
```

Reply:
yea sure, be better this way, done !